src/jit/codegenarm64.cpp

```diff
@@ -3435,42 +3449,45 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
 
     genConsumeOperands(tree);
 
-    emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
-
-    assert(genTypeSize(op1Type) == genTypeSize(op2Type));
-
-    if (varTypeIsFloating(op1Type))
+    if ((tree->gtFlags & GTF_USE_FLAGS) == 0)
     {
-        assert(varTypeIsFloating(op2Type));
-        assert(!op1->isContained());
-        assert(op1Type == op2Type);
+        emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
 
-        if (op2->IsIntegralConst(0))
+        assert(genTypeSize(op1Type) == genTypeSize(op2Type));
+
+        if (varTypeIsFloating(op1Type))
```

Comment:
Does this optimization ever apply to floating point compares? Should the `GTF_USE_FLAGS` check be done only for integer compares perhaps?

Reply:
It does not yet.  I just glanced through the armv8 reference manual.  It looks like there might eventually be cases we could support.  Unless there is an objection, I am happy with it here as it is gated by the setting of the flag.